### DISCUSSION TITLE
Is WAS an application framework in the .NET 5+ implementation?

### DISCUSSION THREAD

**Discussion by learntocode66** [microsoft/WindowsAppSDK#3687] [General] [answered]
I try to understand [.NET implementations](https://learn.microsoft.com/en-us/dotnet/fundamentals/implementations).

> Each implementation of .NET includes the following components:
> 
> (1) One or more runtimes—for example, .NET Framework CLR and .NET 5 CLR.
> (2) A class library—for example, .NET Framework Base Class Library and .NET 5 Base Class Library.
> (3) Optionally, one or more application frameworks—for example, [ASP.NET](https://www.asp.net/), [Windows Forms](https://learn.microsoft.com/en-us/dotnet/desktop/winforms/windows-forms-overview), and [Windows Presentation Foundation (WPF)](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/) are included in .NET Framework and .NET 5+.
> (4) Optionally, development tools. Some development tools are shared among multiple implementations.
>
> There are four .NET implementations that Microsoft supports:
> (1) .NET 5 (and .NET Core) and later versions
> (2) .NET Framework
> (3) Mono
> (4) UWP

Q1. WinUI3 is a part of Windows App SDK (WAS).  Is WAS a class library or an application framework in the .NET 7 implementation?

Q2. In the image below, it seems that the term WinUI3 should better be replaced by "Windows Appp SDK". Is this correct?

Q3. WAS has its own runtime. Is the WAS runtime a part the ".NET CoreCLR"?

[How .NET MAUI works](https://learn.microsoft.com/en-us/dotnet/maui/what-is-maui#how-net-maui-works)
![Image](https://github.com/microsoft/WindowsAppSDK/assets/137069973/1ab62df0-8540-4682-9be5-50d08c5f006d)

**Comment by DarranRowe** [comment, marked as answer]
1) It is closer to a class library. If you want to be extremely precise, the Windows App SDK is a set of Windows Runtime components written in C++. It is then made available in .NET using different means. If you are targeting the .NET Framework, this knows about WinRT components natively. IIRC, early versions of .NET/.NET Core also had this knowledge. But newer versions of .NET use a package like C#/WinRT in order to access it, the Windows App SDK provides the projection as part of the Nuget package so you don't have to make it yourself.

2) This is difficult, but [WinUI 3 is a separate project](https://github.com/microsoft/microsoft-ui-xaml) that is bundled with the Windows App SDK.

3) No. Being a part of the .NET CoreCLR would mean that it would be part of the .NET Runtime.

**Reply by learntocode66** [reply]
Hi @DarranRowe   Thank you! It's very clear about Q1 and Q2.

About Q3, I thought: .NET CoreCLR = .NET CLR = .NET runtime. Is this incorrect?

**Reply by DarranRowe** [reply]
Technically yes. .NET CoreCLR should specifically refer to the native environment, JIT compiler and the like. However, since the CoreCLR ended up being placed in the .NET Runtime repository then it was a bit of an assumption on my part that you meant the .NET Runtime, which also includes the basic cross platform libraries.
If you truly mean .NET CoreCLR, then the answer is still no, since that is not what the .NET CoreCLR is there for. IIRC, the CoreCLR is where the interop related functionality that allows it to call into Windows Runtime components resides though.

**Reply by learntocode66** [reply]
@DarranRowe   Thank you for your help!
